clients/include/testing_bsrxmv.hpp

```diff
@@ -0,0 +1,540 @@
+/* ************************************************************************
+ * Copyright (c) 2021 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * ************************************************************************ */
+
+#pragma once
+#ifndef TESTING_BSRXMV_HPP
+#define TESTING_BSRXMV_HPP
+
+#include "hipsparse.hpp"
+#include "hipsparse_test_unique_ptr.hpp"
+#include "unit.hpp"
+#include "utility.hpp"
+
+#include <cmath>
+#include <hipsparse.h>
+#include <string>
+
+using namespace hipsparse;
+using namespace hipsparse_test;
+
+void testing_bsrxmv_bad_arg(void)
```

Comment:
Should be template<typename T> correct? I think that is what we usually do instead of just testing float.

Reply:
Shouldn't matter for bad args in hipsparse, float is fine.

Reply:
We skip the other types in 33 tests but test all types in the 48 other tests.

Reply:
Ok if we skip them often then its fine here

Reply:
actually much more often since templated bad_args are very often called with <float> only.
I addressed your comment and we call bad arg for the 4 types.

Reply:
fixed